enos/modules/k8s_vault_verify_build_date/main.tf

```diff
@@ -0,0 +1,64 @@
+
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+locals {
+  // build date was introduced in Vault 1.11.0
+  requires_build_date_check = length(regexall("^([2-9])|^(1\\.[1-9][1-9]+)", var.vault_product_version)) > 0
```

Comment:
```suggestion
  requires_build_date_check = length(regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})", var.vault_product_version)) > 0
```

The regex misses cases that may exist in the far future. [long link to example](https://pythex.org/?regex=%5E(%5B2-9%5D)%7C%5E(1%5C.%5B1-9%5D%5B1-9%5D%2B)&test_string=0.0.1%0A1.10.0%0A1.10.12%0A1.11.0%0A1.11.1%0A1.11.12%0A2.11.0%0A3.1.2%0A4.458.80%0A14.0.1%0A54.123.0%0A12341234.000000000.123412334812349rcwhatever&ignorecase=0&multiline=1&dotall=0&verbose=0)

Here is an example of my suggestion: [long link with examples](https://pythex.org/?regex=%5E(%3F%3A1.1%5B%5E0%5D%7C%5B2-9%5D%7C%5Cd%7B2%2C%7D)&test_string=0.0.1%0A1.10.0%0A1.10.12%0A1.11.0%0A1.11.1%0A1.11.12%0A2.11.0%0A3.1.2%0A4.458.80%0A14.0.1%0A54.123.0%0A12341234.000000000.123412334812349rcwhatever&ignorecase=0&multiline=1&dotall=0&verbose=0)

And this is a simple verification that it compiles within Terraform's implementation of PCRE and works as expected:

```
> regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})","1.11.0")
tolist([
  "1.11",
])
> regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})","14.11.0")
tolist([
  "14",
])
> regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})","1.10.12341234")
tolist([])
> regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})","1.10.12341234")
```

Reply:
good catch.